base/docs/basedocs.jl

```diff
@@ -3330,6 +3330,14 @@ unused and delete the entire benchmark code).
     `donotdelete(1+1)`, no add instruction needs to be executed at runtime and
     the code is semantically equivalent to `donotdelete(2).`
 
+!!! note
+    This intrinsic does not affect the semantics of code that is dead because it is
+    *unreachable*. For example, the body of the function `f(x) = false && donotdelete(x)`
+    may be deleted in its entirety. The semantics of this intrinsic only guarantee that
```

Comment:
"For example, in a function like `f(x) = false && donotdelete(x)`, the call to `donotdelete(x)` might be deleted entirely." would be clearer?

Reply:
Eh, I'm not sure it's any better. The point is that the whole branch gets eliminated and the `donotdelete` just happens to be in it, but I don't think either spelling fully conveys that.